controllers/awscluster_controller_test.go

```diff
@@ -314,6 +314,79 @@ func TestAWSClusterReconciler_IntegrationTests(t *testing.T) {
 		_, err = reconciler.reconcileNormal(cs)
 		g.Expect(err.Error()).To(ContainSubstring("The maximum number of VPCs has been reached"))
 	})
+	t.Run("Should successfully delete AWSCluster if creation fails due to VPC limit exceeded", func(t *testing.T) {
```

Comment:
Since this is creating a lot of duplicated code in tests, I think it would be good to just add the deletion part in "Should fail on AWSCluster reconciliation if VPC limit exceeded" test.